src/Appointments.Console/Appointment Management System/Program.cs

```diff
@@ -3,18 +3,22 @@
 using AppointmentManagementSystem.Interfaces;
 using AppointmentManagementSystem.Models;
 using AppointmentManagementSystem.Repositories;
+using AppointmentManagementSystem.Services;
 using Microsoft.Extensions.DependencyInjection;
 
 var serviceProvider = new ServiceCollection()
               .AddSingleton<IManagementRepository<Customer>, CustomerRepository>()
               .AddSingleton<IManagementRepository<Appointment>, AppointmentRepository>()
+              .AddSingleton<IDataEntryService, CustomerDataEntryService>()
+              .AddSingleton<IDataEntryService, AppointmentDataEntryService>()
               .BuildServiceProvider();
 
-var customerRepo = serviceProvider.GetService<IManagementRepository<Customer>>();
-var appointmentRepo = serviceProvider.GetService<IManagementRepository<Appointment>>();
+var services = serviceProvider.GetServices<IDataEntryService>();
+var customerDataEntryService = services.First(o => o.GetType() == typeof(CustomerDataEntryService));
+var appointmentDataEntryService = services.First(o => o.GetType() == typeof(AppointmentDataEntryService));
```

Comment:
This violates the Dependency inversion principle, as the running code depends on the concrete type.